extensions/EXT/EXT_depth_clamp.txt

```diff
@@ -0,0 +1,222 @@
+    clipping to the near and far clip planes and clamping interpolated
+    depth values outside the window-space depth range.  Should the
+    functionality be described this way?
+
+      RESOLUTION:  No.  Describing the functionality as capping is
+      fairly involved.  Eliminating far and near plane clipping and
+      clamping interpolated depth values to the depth range is much
+      simpler to specify.
+
+    Should depth clamping affect points or just line and polygon geometric
+    primitives?
+
+      RESOLUTION:  All geometric primitives are affected by depth
+      clamping.
+
+      In the case of points, if you render a point "in front of" the
+      near clip plane, it should be rendered with the zw value min(n,f)
+      where n and f are the near and far depth range values if depth
+      clamping is enabled.  Similarly, a point "behind" the far clip
+      plane should be rendered with the zw value max(n,f).
+
+    How should the setting of the raster position function when depth
+    clamping is enabled?
+
+      RESOLUTION:  When setting the raster position, clamp the raster
+      position zw to the range [min(n,f),max(n,f)] where n and f are
+      the near and far depth range values.
+
+      This rule ensures that the raster position zw will never be outside
+      the [0,1] range (because n and far are clamped to the [0,1] range).
+      We specify the raster position to be updated this way because
+      otherwise a raster position zw could be specified outside the [0,1]
+      range when depth clamping is enabled, but then if depth clamping
+      is subsequently disabled, that out-of-range raster position zw
+      could not be written to the depth buffer.
+
+      This semantic can make for some unexpected semantics that are
+      described here.  Say that depth clamping is enabled and the raster
+      position is set to point behind the far clip plane such that the
+      pre-clamped zw is 2.5.  Because depth clamping is enabled the
+      raster position zw is clamped to the current near and far depth
+      range values.  Say these values are 0.1 and 0.9.  So 2.5 is clamped
+      to 0.9.
+
+      Now consider what happens if a bitmap (or image rectangle) is
+      rendered with depth testing enabled under various situations.
+      If depth clamping remains enabled and the depth range is unchanged,
+      the bitmap fragments are generated with a zw of 0.9.
+
+      However, if depth range is subsequently set to 0.2 and 0.8 and
+      depth clamping is enabled, the bitmap fragments will have their
+      zw depth component clamped to 0.8.  But if the depth range was
+      changed to 0.2 and 0.8 but depth range clamped is disabled, the
+      bitmap fragments will have a 0.9 zw depth component since then
+      the depth clamping is then not applied.
+
+    What push/pop attrib bits should affect the depth clamp enable?
+
+      RESOLUTION:  GL_ENABLE_BIT and GL_TRANSFORM_BIT.
```

Comment:
This issue not applicable for ES. 

What I'd suggest instead of repeating all the issues from the earlier versions of the extension, is just say
"See the issue list in GL_ARB_depth_clamp" and only include issues unique to this extension. 

It's also a good idea to include an issue "How does this extension differ from ARB_depth_clamp", and describe any differences (of which there aren't much). 

In newer extensions, the issues list also goes at the end, just before the revision history.

Reply:
changed accordingly.